sparkle/src/components/DataTable.tsx

```diff
@@ -25,7 +25,12 @@ interface DataTableProps<TData, TValue> {
   initialColumnOrder?: SortingState;
 }
 
-export function DataTable<TData, TValue>({
+interface TBaseData {
+  onClick?: () => void;
+  onMoreClick?: () => void;
+}
+
+export function DataTable<TData extends TBaseData, TValue>({
```

Comment:
On line 19 you need to specify extends TBaseData so it's enforced everywhere.

Reply:
🙏🏼